### ISSUE TITLE
Extract the `initial_state` from the ChainConfig

### ISSUE THREAD

**Issue by xgreenx** [FuelLabs/fuel-core#1450]
## Overview

Extract the initial_state from the `ChainConfig`. We still need it during the start of the network, so this information should be provided along with `--chain` argument. We want to store an initial state in a separate folder. 

We have 3 flows:
- Contracts
- Messages
- UTXOs

Each flow should have its own file with the state. It allows us parallelize initialing from the snapshot in the future.

## Implementation details

The `init_height` or `base_height` should be static part of `ChainConfig`.
A new `--chain-initia-state` arguments accepts path to the folder, where initial state is stored(Not to the file). it should be part of the `fuel-core run` command.
The `import_genesis_block` function should split importing of each flow into a separate task and do in parallel. Each task should return unordered vector of merkle roots. Before calculating the final root for each flow we need to sort all roots and create a final BMT root after.

**Comment by xgreenx** [comment]
Implemented during https://github.com/FuelLabs/fuel-core/pull/1459. The part related to parallelization will be done in a separate PR.
